tensorflow_quantum/core/src/circuit_parser_qsim.cc

```diff
@@ -0,0 +1,373 @@
+}
+
+Status XXGate(const Operation& op, const SymbolMap& param_map,
+              const unsigned time, QsimCircuit* circuit) {
+  return TwoEigenGate(op, param_map, &qsim::Cirq::XXPowGate<float>::Create,
+                      time, circuit);
+}
+
+Status YGate(const Operation& op, const SymbolMap& param_map,
+             const unsigned time, QsimCircuit* circuit) {
+  return SingleEigenGate(op, param_map, &qsim::Cirq::YPowGate<float>::Create,
+                         time, circuit);
+}
+
+Status YYGate(const Operation& op, const SymbolMap& param_map,
+              const unsigned time, QsimCircuit* circuit) {
+  return TwoEigenGate(op, param_map, &qsim::Cirq::YYPowGate<float>::Create,
+                      time, circuit);
+}
+
+Status ZGate(const Operation& op, const SymbolMap& param_map,
+             const unsigned time, QsimCircuit* circuit) {
+  return SingleEigenGate(op, param_map, &qsim::Cirq::ZPowGate<float>::Create,
+                         time, circuit);
+}
+
+Status ZZGate(const Operation& op, const SymbolMap& param_map,
+              const unsigned time, QsimCircuit* circuit) {
+  return TwoEigenGate(op, param_map, &qsim::Cirq::ZZPowGate<float>::Create,
+                      time, circuit);
+}
+
+Status CZGate(const Operation& op, const SymbolMap& param_map,
+              const unsigned time, QsimCircuit* circuit) {
+  return TwoEigenGate(op, param_map, &qsim::Cirq::CZPowGate<float>::Create,
+                      time, circuit);
+}
+
+Status CXGate(const Operation& op, const SymbolMap& param_map,
+              const unsigned time, QsimCircuit* circuit) {
+  return TwoEigenGate(op, param_map, &qsim::Cirq::CXPowGate<float>::Create,
+                      time, circuit);
+}
+
+Status SwapGate(const Operation& op, const SymbolMap& param_map,
+                const unsigned time, QsimCircuit* circuit) {
+  return TwoEigenGate(op, param_map, &qsim::Cirq::SwapPowGate<float>::Create,
+                      time, circuit);
+}
+
+Status ISwapGate(const Operation& op, const SymbolMap& param_map,
+                 const unsigned time, QsimCircuit* circuit) {
+  return TwoEigenGate(op, param_map, &qsim::Cirq::ISwapPowGate<float>::Create,
+                      time, circuit);
+}
+
+// single qubit PhasedXPow -> Create(time, q0, pexp, exp)
+inline Status PhasedXGate(const Operation& op, const SymbolMap& param_map,
+                          const unsigned time, QsimCircuit* circuit) {
+  int q0;
+  bool unused;
+  float pexp, pexp_s, exp, exp_s, gs;
+  Status u;
+  unused = absl::SimpleAtoi(op.qubits(0).id(), &q0);
+
+  u = ParseProtoArg(op, "exponent", param_map, &exp);
+  if (!u.ok()) {
+    return u;
+  }
+  u = ParseProtoArg(op, "exponent_scalar", param_map, &exp_s);
+  if (!u.ok()) {
+    return u;
+  }
+  u = ParseProtoArg(op, "phase_exponent", param_map, &pexp);
+  if (!u.ok()) {
+    return u;
+  }
+  u = ParseProtoArg(op, "phase_exponent_scalar", param_map, &pexp_s);
+  if (!u.ok()) {
+    return u;
+  }
```

Comment:
ditto. `exponent` vs `exponent_scalar` and `phase_exponent` vs `phase_exponent_scalar`.